src/game/game.cpp

```diff
@@ -6064,7 +6064,7 @@ bool Game::combatBlockHit(CombatDamage &damage, std::shared_ptr<Creature> attack
 			if (secondaryReflectPercent > 0 || secondaryReflectFlat > 0) {
 				if (!canReflect) {
 					damageReflected.primary.type = damage.secondary.type;
-					damageReflected.primary.value = std::ceil(damage.secondary.value * secondaryReflectPercent / 100.) + std::max(-static_cast<int32_t>(std::ceil(attacker->getMaxHealth() * 0.01)), std::max(damage.secondary.value, -(static_cast<int32_t>(secondaryReflectFlat))));
+					damageReflected.primary.value = std::ceil(damage.secondary.value * (int)secondaryReflectPercent / 100.) + std::max(-static_cast<int32_t>(std::ceil(attacker->getMaxHealth() * 0.01)), std::max(damage.secondary.value, -(static_cast<int32_t>(secondaryReflectFlat))));
```

Comment:
Are you sure it shouldn't be (double) since it is dividing by a double?

Reply:
by using double or int in that code the multiplier is strange, seems are giving more damage than the correct..
![image](https://github.com/opentibiabr/canary/assets/10325310/2e6d698c-e82d-4393-bba3-c5ada469586c)

Reply:
@dudantas I noticed that this error of reflecting 42 million healing only applies to secondary damage types. There is no casting required in the part of the code where primary damage type's reflect is calculated. Thats because the "primaryReflectPercent" there is a signed integer. 

So in the secondary dmg scope i removed casting, and instead changed "primaryReflectPercent" to signed int (its in the 2nd commit). From the tests i did, it now works the same as with first commit.

Reply:
@Schiffers I built a fresh canary server (without my pr), and noticed that runes and wands also have the problem you present. I don't really play real tibia, so i don't have idea whether reflected damage should be calculated based on damage attacker rolled or the actual damage monster received. If its the first case, then its coded correctly. If its seconds case, then its in need of a fix.

I did quick maths, and for your situation on screen damage attacker rolled would be ~686 (206 * 0.3^-1), and reflected damage should be ~912 (686 * 1.33). On my server, the calculation is correct (https://i.imgur.com/upwlg7G.png). I bet you must have damage reduction from items/wheel.

Also, in the canary server, spectres have reflect factor of 133%, but tibia fandom states 166%, and br wiki states >=150%, so idk which is correct.

Reply:
The thing is
The reflect damage is calculated by the type of damage 
The system should consider only ice damage for ice damage reflection
The maths you did seems strange... 
I'm causing 206 ice damage
And get reflected 815
Considering the set and the mitgation of the player.... 
How it's look correct? 
Will do another test with the last commit

Reply:
update...
I couldn't get the point before during my tests ....
burster spectre have 70% resist of ice damage, so because of this resistence my damage was too low and the reflect come "strong" ....
**BUT**
the damage received my reflect damage should be base in the **DAMAGE CAUSED** and not the base...